src/people.rs

```diff
@@ -310,6 +523,15 @@ mod test {
             0
         }
     });
+    define_derived_property!(TrailRunner, bool, [IsRunner, Age], |is_runner, age| {
+        is_runner && age > 29
+    });
+    define_derived_property!(
+        UltraRunner,
+        bool,
+        [TrailRunner, Age],
```

Comment:
I know I'm nitpicking, but I think it would be better to have properties which were self-evident from the name,  and it's not obvious that a trailrunner is a runner > 29.